modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/GranuleDescriptor.java

```diff
@@ -137,14 +140,25 @@
 
     OverviewsController overviewsController;
 
+    private CoordinateReferenceSystem granuleCRS;
```

Comment:
Ok, I'll take a look at getting the CRS from the Envelope instead. Probably a better place to get it.